akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -247,6 +255,10 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   private val restartCounter = new RestartCounter(maxRestarts, restartTimeout)
 
   val envelopePool = new EnvelopeBufferPool(ArteryTransport.MaximumFrameSize, ArteryTransport.MaximumPooledBuffers)
+  val largeEnvelopePool: Option[EnvelopeBufferPool] =
+    if (largeMessageDestinationsEnabled) Some(new EnvelopeBufferPool(ArteryTransport.MaximumLargeFrameSize, ArteryTransport.MaximumPooledBuffers))
+    else None
```

Comment:
why not just create the pool anyway? It will allocate buffers on demand anyway, the main cost will be its internal queue but that should not be to large probably anyway.